app/util/analytics.py

```diff
@@ -114,24 +125,63 @@ def get_events(since_timestamp=None, since_id=None):
     since_timestamp = float(since_timestamp) if since_timestamp else 0.0
     since_id = int(since_id) if since_id else 0
 
-    with open(_eventlog_file, 'r') as f:
-        # todo(joey): This is inefficient since it reads the whole log file into memory (can be several megabytes).
-        # We probably want something like http://code.activestate.com/recipes/120686/
-        reversed_log_lines = reversed(f.readlines())
+    filtered_events_from_cache = get_events_from_cache(since_timestamp, since_id)
+    if len(filtered_events_from_cache) > 0:
```

Comment:
Sounds good. Yeah we could just make it load the file any time the number of events returned from the cache equals `LOG_CACHE_MAX_SIZE` -- the annoying part is de-duping cached events and events from file, and also dealing with those file log rotations.

If you decide to make these improvements in pieces / multiple PRs, that's cool too. What you have now is still an improvement over my dumb implementation. :)